File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.cpp
class Solution {
public:
    vector<string> findRepeatedDnaSequences(string s) {
        int n = s.size();
        unordered_map<string, int> mp;

        for (int i = 0; i < n; i++) {
            string temp = s.substr(i, 10);
            mp[temp]++;
        }

        vector<string> ans;
        for (auto& p : mp) {
            if (p.second > 1)
                ans.push_back(p.first);
        }

        return ans;
    }
};

File: 0187-repeated-dna-sequences/0187-repeated-dna-sequences.java
class Solution {
    public List<String> findRepeatedDnaSequences(String s) {
        int n = s.length();

        HashMap<String, Integer> mp = new HashMap<>();

        for (int i = 0; i <= n - 10; i++) {
            String temp = s.substring(i, i + 10);
            mp.put(temp, mp.getOrDefault(temp, 0) + 1);
        }

        ArrayList<String> ans = new ArrayList<>();

        for (Map.Entry<String, Integer> p : mp.entrySet()) {
            if (p.getValue() > 1) {
                ans.add(p.getKey());
            }
        }

        return ans;
    }
}